### PR TITLE
fix import when a full disk has zero partitions

### PR DESCRIPTION
**Pull request by charlessuh**

### CODE REVIEW

File: lib/libzpool/kernel.c [hunk 1]
```diff
@@ -813,11 +813,9 @@ random_get_pseudo_bytes(uint8_t *ptr, size_t len)
 int
 ddi_strtoull(const char *str, char **nptr, int base, u_longlong_t *result)
 {
-	(void) nptr;
-	char *end;
-
+	char* end = NULL;
 	*result = strtoull(str, &end, base);
-	if (*result == 0)
```
Comment (charlessuh) [inline lib/libzpool/kernel.c:820]: It's not necessarily an error for `strtoull` to return zero, if it actually parsed a zero from the string (e.g. `0 1 2 3`.

File: lib/libzutil/os/windows/zutil_import_os.c [hunk 2]
```diff
@@ -621,10 +623,13 @@ zpool_find_import_blkid(libpc_handle_t *hdl, pthread_mutex_t *lock,
 
 				slice = zutil_alloc(hdl, sizeof (rdsk_node_t));
 
+				uint64_t size = GetFileDriveSize(disk);
+
 				error = asprintf(&slice->rn_name,
 				    "#%llu#%llu#%s",
-				    0ULL, GetFileDriveSize(disk),
+				    0ULL, size,
```
Comment (charlessuh) [inline lib/libzutil/os/windows/zutil_import_os.c:630]: I appear to be getting some funny business where this gets truncated to INT_MAX (so I get paths starting with `#0#2147483648`) unless you are more explicit.

File: lib/libzpool/kernel.c (hunk 1 above)
Comment (lundman) [reply lib/libzpool/kernel.c:820]: hah yep, you are right

File: lib/libzutil/os/windows/zutil_import_os.c (hunk 2 above)
Comment (lundman) [reply lib/libzutil/os/windows/zutil_import_os.c:630]: could be missing prototype for GetFileDriveSize(), defaults to int

**Review by lundman — APPROVED** [review]

**Pull request merged**